### PR TITLE
Optimizes, bugfixes for NTSL; adds two new channel vars

### PR DESCRIPTION
**Pull request by Altoids1**
### Making NTSL Not Be Shite: Part 1

This is mostly just a packet of minor additions/bugfixes done during a quick scan of the code I did yesterday. All of them have been personally tested, pinky promise.

#### Changelog

:cl:  
rscadd: Added two new on-boot NTSL variables, $aiprivate and $centcom
bugfix: NTSL will now properly handle having $content be set to a number value.
bugfix: NTSL should now parse comments correctly.
bugfix: Optimized how NTSL deals with numeric operators and comments. Should run/compile 1% faster or something. *shrug
spellcheck: Fixed minor typoes within some of the telecomms computers
/:cl:

### CODE REVIEW

File: code/modules/scripting/Implementations/Telecomms.dm [hunk 1]
```diff
@@ -316,19 +324,26 @@ var/const/MAX_MEM_VARS	 = 500
...
 	if(findtext(num2text(freq), ".")) // if the frequency has been set as a decimal
-		freq *= 10 // shift the decimal one place
-
-	if(!job)
-		job = "Unknown"
+		//freq *= 10 // shift the decimal one place
+		////Except that we don't trust floating-point arithmetic--
+		//// woah, deja vu.
+		var/shittyfreq = num2text(freq)
+		freq = text2num(copytext(shittyfreq, 1, findtext(shittyfreq,".")) + copytext(shittyfreq,-1))
```
Comment (Cruix) [inline code/modules/scripting/Implementations/Telecomms.dm]: Floating point multiplication by whole numbers is perfectly safe and much faster than string operations.

Comment (Altoids1) [reply code/modules/scripting/Implementations/Telecomms.dm]: I have personally had issues before with broadcast() and signal() failing when given decimal inputs, and figured this was the reason. I think you'd have to be right, though. Will fix.

**Comment by Ratvar** [comment]
NTSL can fuck centcom radio now? You sure?

**Comment by Amelia0010** [comment]
it's been able to the entire ~~Station~~ time

**Comment by Partheo** [comment]
You are great

File: code/modules/scripting/Scanner/Scanner.dm [hunk 2]
```diff
@@ -244,45 +242,30 @@
...
+*/
 
-				if(comm == 2)
+		ReadComment(isshort)
+			// Remember that we still have that $codepos pointer variable,
+			// since it was a global
+			if(isshort) // If line comment
+				++codepos // To skip over the second slash
+				while(++codepos>lentext(code))
+				{
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline code/modules/scripting/Scanner/Scanner.dm]: No

Comment (Altoids1) [reply code/modules/scripting/Scanner/Scanner.dm]: No? What'cha mean by no? 🤔 
I mean, yeah, I could've done ``for(;codepos>lentext(code);++codepos)``, but I don't think that would have been any more clear than just using the while loops that were there in the original version of this function.

Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [reply code/modules/scripting/Scanner/Scanner.dm]: By no I meant no braces. I know the interpreter code is garbage but don't make it even worse.

Comment (Altoids1) [reply code/modules/scripting/Scanner/Scanner.dm]: There's nothin' in the style guide against it. In fact, I think it's more readable when dealing with large amounts of nesting, so, unless a headcoder gets into a fuss about it, it's staying.

~~Go back to Python you tab-lovin' pieces of shi~~

Comment (Amelia0010) [reply code/modules/scripting/Scanner/Scanner.dm]: I'm making a fuss of it, and I know @AshCorr will too, and she's the only one currently merging. Remove them or this probably won't be merged 🤔

**Comment by Amelia0010** [comment]
👍

**Comment by Amelia0010** [comment]
Apparently I can make people do what I want by threatening them with Ash. Who knew?

**Comment by Altoids1** [comment]
MERGE ME BB

**Comment by AsV9** [comment]
What could possibly go wrong?

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
did you check for any hidden braces

**Pull request merged**